amy/trainings/views.py

```diff
@@ -125,18 +126,45 @@ def all_trainees(request):
         # Bulk add progress to selected trainees
         form = BulkAddTrainingProgressForm(request.POST)
         if form.is_valid():
+            errors = []
             for trainee in form.cleaned_data["trainees"]:
-                TrainingProgress.objects.create(
-                    trainee=trainee,
-                    requirement=form.cleaned_data["requirement"],
-                    state=form.cleaned_data["state"],
-                    event=form.cleaned_data["event"],
-                    url=form.cleaned_data["url"],
-                    notes=form.cleaned_data["notes"],
+                try:
+                    progress = TrainingProgress(
+                        trainee=trainee,
+                        requirement=form.cleaned_data["requirement"],
+                        involvement_type=form.cleaned_data["involvement_type"],
+                        state=form.cleaned_data["state"],
+                        event=form.cleaned_data["event"],
+                        url=form.cleaned_data["url"],
+                        date=form.cleaned_data["date"],
+                        notes=form.cleaned_data["notes"],
+                    )
+                    progress.full_clean()
+                    progress.save()
+                except ValidationError as e:
+                    errors.append(e)
+
+            if len(errors) > 0:
```

Comment:
Idiomatic Python version:
```suggestion
            if errors:
```